engine/access/rpc/backend/connection_factory.go

```diff
@@ -84,6 +108,17 @@ func (cf *ConnectionFactoryImpl) createConnection(address string, timeout time.D
 		Timeout: timeout,
 	}
 
+	var connInterceptors []grpc.UnaryClientInterceptor
+
+	if cf.ConnectionsCache != nil {
+		if res, ok := cf.ConnectionsCache.Get(address); ok {
+			cachedClient := res.(*CachedClient)
+			connInterceptors = append(connInterceptors, createRequestWatcherInterceptor(cachedClient))
+		}
+	}
```

Comment:
in the cases where there is a cache, the client has already been retrieved. rather than a second cache lookup here, how about passing in a `*CachedClient`. If there's no cache, the value would just be nil